mne_bids/read.py

```diff
@@ -154,7 +154,7 @@ def _read_events(events_data, event_id, raw, verbose=None):
         verbose=verbose
     )
     all_dur = raw.annotations.duration
-    if all_events.size == 0 and 'rest' not in raw.filenames[0]:
+    if all_events.size == 0:
```

Comment:
Why did you change this? :)
This was here specifically to suppress a warning in case of resting-state data

Reply:
This was causing an error when `raw.filename` did not exist.

Searching for 'rest' in file name seems to be a dubious way of checking for resting state data. It would believe any subject, session, or task with string 'rest' to be resting-state data (e.g. watching clips from "Forrest Gump"). On the other hand, non-resting data with no events are also possible.

Reply:
It's clear that the existing heuristic for detecting resting-state data could be improved, but it was added because there was an actual demand / use case. So please either keep it in or try to make it better, but don't just remove it.

Reply:
Changing `_read_events` function seemed dangerous, so I have improved the warning based on BIDS recommendations.